### PR TITLE
New command to look at individual tarot cards.

### PR DESCRIPTION
**Pull request by h-anjru**

### CODE REVIEW

File: crimsobot/cogs/mystery.py [hunk 1]
```diff
@@ -58,6 +58,88 @@ async def tarot(self, ctx: commands.Context, spread: str = 'ppf') -> None:
...
+        # wait for user to spcify suit
+        msg = await self.bot.wait_for('message', check=suit_check, timeout=20)
+
+        if msg is None:
+            await prompt_suit.delete()
+            return
+        else:
+            suit_choice = int(msg.content)
+            await msg.delete()
+            await prompt_suit.delete()
```
Comment (williammck) [inline crimsobot/cogs/mystery.py:101]: nit: I'm a fan of returning early in situations like this ([see this for examples](https://blog.timoxley.com/post/47041269194/avoid-else-return-early)). Makes the code flatter and cleaner IMO.

```suggestion
        await prompt_suit.delete()

        if msg is None:
            return

        suit_choice = int(msg.content)
        await msg.delete()
```

File: crimsobot/cogs/mystery.py [hunk 2]
```diff
@@ -58,6 +58,88 @@ async def tarot(self, ctx: commands.Context, spread: str = 'ppf') -> None:
...
 
+    @commands.command(aliases=['tarotcard'])
+    async def card(self, ctx: commands.Context) -> None:
+        """Inspect a tarot card chosen from the menu for further study."""
+
+        # the suits
+        suit_dict = {0: 'Major arcana', 1: 'Wands', 2: 'Pentacles', 3: 'Cups', 4: 'Swords'}
+        suit_list = []
+        for suit in suit_dict:
+            suit_list.append('{}. {}'.format(suit, suit_dict[suit]))
```
Comment (williammck) [inline crimsobot/cogs/mystery.py:69]: not really sure why this is a dict?
would recommend something like this:
```suggestion
        suits = ['Major arcana', 'Wands', 'Pentacles', 'Cups', 'Swords']
        suit_list = []
        for idx, suit in enumerate(suits):
            suit_list.append('{}. {}'.format(idx, suit))
```
keeping in mind you'll need to change any later references of `suit_dict` to `suits`

File: crimsobot/cogs/mystery.py [hunk 3]
```diff
@@ -58,6 +58,88 @@ async def tarot(self, ctx: commands.Context, spread: str = 'ppf') -> None:
...
+        # wait for user to spcify suit
+        msg = await self.bot.wait_for('message', check=card_check, timeout=20)
+
+        if msg is None:
+            await prompt_card.delete()
+            return
+        else:
+            card_choice = int(msg.content)
+            await msg.delete()
+            await prompt_card.delete()
```
Comment (williammck) [inline crimsobot/cogs/mystery.py:138]: same comment here as 95-101.

File: crimsobot/utils/tarot.py [hunk 4]
```diff
@@ -10,6 +10,41 @@
 from crimsobot.utils.tools import clib_path_join
 
 
+def list_cards(suit: Optional[str]) -> List[str]:
+    cards_in_suit = []
+
+    if suit == 'Major arcana':
```
Comment (williammck) [inline crimsobot/utils/tarot.py:16]: i'm fine with this for right now but we should really have a better way of identifying what suit something is in other than name or position in the deck.

File: crimsobot/cogs/mystery.py [hunk 5]
```diff
@@ -58,6 +58,88 @@ async def tarot(self, ctx: commands.Context, spread: str = 'ppf') -> None:
...
+        else:
+            suit_choice = int(msg.content)
+            await msg.delete()
+            await prompt_suit.delete()
+
+        # prompt 2 of 2: choose card in suit
+        card_names = tarot.list_cards(suit_dict[suit_choice])
+        card_list = []
+        for i in range(0, len(card_names)):
+            card_list.append('{}. {}'.format(i+1 if suit_choice != 0 else i, card_names[i]))
```
Comment (williammck) [inline crimsobot/cogs/mystery.py:107]: i feel like this is cleaner:
```suggestion
        for idx, card_name in enumerate(card_names):
            idx = idx + 1 if suit_choice > 0 else idx
            card_list.append('{}. {}'.format(idx, card_name))
```

File: crimsobot/cogs/mystery.py [hunk 6]
```diff
@@ -58,6 +60,92 @@ async def tarot(self, ctx: commands.Context, spread: str = 'ppf') -> None:
...
+                return False
+
+        # wait for user to spcify suit
+        try:
+            msg = await self.bot.wait_for('message', check=suit_check, timeout=20)
+        except asyncio.TimeoutError:
+            await prompt_suit.delete()
+            return
+
+        if msg is not None:
```
Comment (williammck) [inline crimsobot/cogs/mystery.py:102]: Is there a point where this would be `None` now that we’re catching TimeoutError? Same for line 141 below.

Comment (h-anjru) [reply crimsobot/cogs/mystery.py:102]: Hell if I know! Should this be amended?

File: crimsobot/cogs/mystery.py [hunk 7]
```diff
@@ -58,6 +60,92 @@ async def tarot(self, ctx: commands.Context, spread: str = 'ppf') -> None:
...
+            suit_choice = int(msg.content)
+            await msg.delete()
+            await prompt_suit.delete()
```
Comment (williammck) [inline crimsobot/cogs/mystery.py:105]: Let's change this to the following just to be safe. Same around 141-144
```suggestion
        await prompt_suit.delete()

        if msg is None:
            return

        suit_choice = int(msg.content)
        await msg.delete()
```

Comment (h-anjru) [reply crimsobot/cogs/mystery.py:105]: Alright will do.

**Pull request merged**